bitcoin/Cargo.toml

```diff
@@ -16,13 +16,15 @@ exclude = ["tests", "contrib"]
 [features]
 default = [ "std", "secp-recovery" ]
 std = ["base58/std", "bech32/std", "hashes/std", "hex/std", "internals/std", "io/std", "secp256k1/std", "units/std"]
-rand-std = ["secp256k1/rand-std", "std"]
-rand = ["secp256k1/rand"]
 serde = ["actual-serde", "hashes/serde", "secp256k1/serde", "internals/serde", "units/serde"]
 secp-lowmemory = ["secp256k1/lowmemory"]
 secp-recovery = ["secp256k1/recovery"]
 bitcoinconsensus-std = ["bitcoinconsensus/std", "std"]
 
+# These features are additive but in typical usage you just want to pick one, ether std or no-std.
```

Comment:
Just a point that adding stuff as a comment in a `Cargo.toml` is fine,
but where most users would check is the documentation.

What about adding something here?
https://github.com/rust-bitcoin/rust-bitcoin/blob/aedb09745a5b441c5dafa3ea10e3a7658c4d402c/bitcoin/src/lib.rs#L16-L30

Reply:
Ah good point. On that topic, do you know if this this way of documenting the features is standard practice, it always seemed ad hoc to me?

Reply:
I'll fix this PR to include it as suggested though.

Reply:
The rustdocs build system gives a warning (visible on the [features page](https://docs.rs/crate/bitcoin/latest/features) of `rust-bitcoin` docs. But I couldn't find how to do any better than what we are doing?

Reply:
Yeah, weird, I wonder what kind of "structured metadata" we are supposed to have.

Reply:
We are the same with `serde`, they have the same warning:
https://docs.rs/crate/serde/latest/features